### PR TITLE
feat: log events from account-kit packages

### PR DESCRIPTION
**Pull request by moldy530**
- **feat: log infra events**
- **feat: log events from core package**
- **feat: log events from signer package**
- **feat: add react events logging**
- **feat: add signer profiling metrics**

# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR focuses on enhancing the logging functionality across various modules by integrating event tracking and updating version dependencies. It introduces new metrics, modifies existing logging methods, and fetches remote configurations.

### Detailed summary
- Updated `VERSION` from `"4.0.0-beta.8"` to `"4.0.0-beta.9"`.
- Added dependency on `@account-kit/logging` in `package.json`.
- Introduced `fetchRemoteWriteKey` function in `_writeKey.ts`.
- Enhanced event tracking in multiple components and hooks.
- Updated metrics schema in `infra/src/metrics.ts` and `signer/src/metrics.ts`.
- Replaced existing logging with `ReactLogger` and `CoreLogger` in various hooks and components.
- Improved error handling and analytics write key logic in `client.ts`.
- Removed hardcoded write key in `postbuild.ts` and adjusted CI handling.
- Refactored `BaseAlchemySigner` methods to include logging for authentication and transaction processes.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: doc-gen/src/commands/generate.ts [hunk 1]
```diff
@@ -193,9 +193,10 @@ function generateClassDocs(
       ts.isConstructorDeclaration(member) ||
       ts.isMethodDeclaration(member) ||
       (ts.isPropertyDeclaration(member) &&
-        member.initializer &&
-        (ts.isArrowFunction(member.initializer) ||
-          ts.isFunctionExpression(member.initializer)))
+        ((member.initializer &&
+          (ts.isArrowFunction(member.initializer) ||
+            ts.isFunctionExpression(member.initializer))) ||
+          (member.type && ts.isFunctionLike(member.type))))
```
Comment (moldy530) [inline doc-gen/src/commands/generate.ts:199]: this is to handle the `SignerLogger.profiled` class property declarations which are still functions. This allows us to fallback on to the type defintion if it exists so we can still generate docs

**Review by rthomare — APPROVED** [review]

File: account-kit/core/src/actions/createAccount.ts [hunk 2]
```diff
@@ -112,6 +123,15 @@ export async function createAccount<TAccount extends SupportedAccountTypes>(
           signer,
           transport: (opts) => transport({ ...opts, retryCount: 0 }),
           chain,
+        }).then((account) => {
```
Comment (rthomare) [inline account-kit/core/src/actions/createAccount.ts:126]: Question? Why not put this directly in the createXXXXAccount functions the smart accounts sub package?

Comment (moldy530) [reply account-kit/core/src/actions/createAccount.ts:126]: This particular function is used to create accounts leveraging our signer and handles writing to state as well. Later will make up part of the API for one contract

Comment (moldy530) [reply account-kit/core/src/actions/createAccount.ts:126]: OH I'm dumb. I see what you mean

**Pull request merged**